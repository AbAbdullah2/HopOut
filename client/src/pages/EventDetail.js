import React, { useEffect, useState } from 'react';
import { useParams, useNavigate } from 'react-router-dom'
import Header from '../components/Header';
import EventHostView from '../components/EventHostView';
import DeleteEventConfirm from '../components/DeleteEventConfirm';
import { formatEventDates } from '../helpers/FormatDate';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { solid } from '@fortawesome/fontawesome-svg-core/import.macro'
import { getEvent, getUser, getAllUsers } from '../services/api';
import toast, { Toaster } from 'react-hot-toast';
import { rsvpToEvent, cancelRsvp } from '../services/api';
import { Table, Dropdown } from 'flowbite-react';

export default function EventDetail(props) {
 const {eventid} = useParams();
 const {curUser, setCurUser} = props;

 const [event, setEvent] = useState(null);
 const [host, setHost] = useState(null);
 const [showConfirm, setShowConfirm] = useState(false);

 const [atCapacity, setAtCapacity] = useState();
 const [rsvp, setRsvp] = useState(curUser.attending.includes(eventid));
 const [attendeesCount, setAttendeesCount] = useState(0);
 const [attendees, setAttendees] = useState([]);
 const [attendeesAndInvitees, setAttendeesAndInvitees] = useState([]);
 const [invitees, setInvitees] = useState([]);
 const [checkedArray, setCheckedArray] = useState([true, true, true]);

 const navigate = useNavigate();
 useEffect(() => {
   if (!curUser || curUser === null) navigate('/login');
   getEvent(eventid).then((res) => {
     setEvent(res.data.data);
     setAtCapacity(res.data.data.capacity === res.data.data.attendees.length);
     setAttendeesCount(res.data.data.attendees.length);
   }); 
 }, [curUser, navigate, eventid]);
 
 useEffect(() => {
  getUser(curUser._id).then(userData => {
    setRsvp(userData.data.data.attending.includes(eventid));
})
}, [curUser, eventid]);

useEffect(() => {
  const getPeople = async () => {
    const response = await getAllUsers();
    const users = response.data.data;
    const att = users.filter((user) => {return user.attending.includes(eventid)});
    setAttendees(att);
    const ppl = users.filter((user) => {return user.attending.includes(eventid) || user.invited.includes(eventid)});
    setAttendeesAndInvitees(ppl);
    const inv = users.filter((user) => {return user.invited.includes(eventid)});
    setInvitees(inv);
  }
  getPeople();
}, [curUser, eventid]);

useEffect(() => {
  const getPeople = async () => {
    const response = await getAllUsers();
    const users = response.data.data;
  }
  getPeople();
}, [curUser, eventid]);

 useEffect(() => {
   if (event !== null) {
     getUser(event.organizer).then((res) => {
       setHost(res.data.data);
     });
   }
 }, [event, eventid]);

 const toShow = (user) => {
  if (!checkedArray[0] && (attendees.includes(user) && invitees.includes(user))) {
    return false;
  }
  if (!checkedArray[1] && (!attendees.includes(user) && invitees.includes(user))) {
    return false;
  }
  if (!checkedArray[2] && (attendees.includes(user) && !invitees.includes(user))) {
    return false;
  }
  return true;
 }

 const confirmRsvp = () => {
   // set rsvp state to true, show toast message and rsvp in the backend
   setRsvp(true);
   rsvpToEvent(curUser._id, eventid);
   toast.success('Successfully RSVP\'d to this event!');
   curUser.attending.push(eventid);
   setAttendeesCount(attendeesCount + 1);
 }

 const cancelRsvpHelper = () => {
   // set rsvp state to false, show toast message and cancel rsvp in the backend
   setRsvp(false);
   cancelRsvp(curUser._id, eventid);
   toast.error('Your RSVP has been cancelled!');
   curUser.attending.pop(eventid);
   setAttendeesCount(attendeesCount - 1);
 }


 return event === null ? '' : (
   <div className='bg-stone-100 min-h-screen'>
     <Toaster />
     <div className='mx-auto flex flex-col h-full'>
       <DeleteEventConfirm curUser={curUser} setCurUser={setCurUser} eventid={eventid} showConfirm={showConfirm} setShowConfirm={setShowConfirm}/>
       <Header icons={true} curUser={curUser} setCurUser={setCurUser}/>
       <img src={event.coverId} alt={event.title} className='w-full object-cover h-80' />
       <div className="relative">
          <img src={event.coverId} alt={event.title} className='w-full object-cover h-60' />
          { curUser.organizing && curUser.organizing.includes(eventid) ? 
            <EventHostView eventid={eventid} setShowConfirm={setShowConfirm} /> :
            <></> }
       </div>
       <div className='m-5'>
         <p className='text-4xl font-extrabold text-center'>{event.name}</p>
         <p className='text-lg my-2 text-center'><FontAwesomeIcon icon={solid('calendar')} /> {formatEventDates(new Date(event.start), new Date(event.end))}</p>
         <p className='text-lg my-2 text-center'><FontAwesomeIcon icon={solid('location-dot')} /> {event.location.address}, {event.location.city}, {event.location.state} {event.location.zip}</p>
         <p className='text-lg my-2 text-center'><FontAwesomeIcon icon={solid('user')} /> Organized by <a href={host ? "/profile/"+host._id : "/profile/"}>{host ? host.name : ''}</a></p>
         <p className='text-lg my-2 text-center'><FontAwesomeIcon icon={event.visibility === 'public' ? solid('eye') : solid('eye-slash')} /> {event.visibility === 'public' ? "Public Event" : "Private Event"}</p>
         <p className='text-lg my-2 text-center'><FontAwesomeIcon icon={solid('users')} /> {attendeesCount} attending</p>
         <p className='text-lg my-2 text-center'><FontAwesomeIcon icon={solid('circle-exclamation')} /> {event.capacity - attendeesCount} spots remaining</p>
         {
         event.organizer === curUser._id ?
         <div></div> :
         <div>
          {
          rsvp ?
            <div className='flex flex-col items-center'>
            <button className="text-2xl bg-transparent hover:bg-blue-500 text-blue-700 font-semibold hover:text-white py-2 px-4 border border-blue-500 hover:border-transparent rounded" onClick={cancelRsvpHelper}>
              Cancel RSVP
            </button>
            </div>     
          :
          <div>
            { !atCapacity ?
              <div className='flex flex-col items-center'>
                <button className="text-2xl bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded" onClick={confirmRsvp}>
                  RSVP
                </button>
              </div>
              :
              <div className='flex flex-col items-center'>
                <p>This Event Has Reached Its Capacity!</p>
              </div>
            }
            </div>
          }
         </div>
         }
         <div className="flex items-center justify-center mt-4">
           {event.categories.map((c, i) => {
             return <div key={i} className="bg-gray-400 p-4 rounded-full items-center leading-none w-fit lg:rounded-full flex lg:inline-flex mr-2">{c}</div>
           })}
         </div>
         <hr className='my-4 bg-stone-800 h-1' />
        
         <p className='my-2'>{event.description}</p>   
         {event.organizer !== curUser._id ? 
          <div>
          <p>Attendees</p>
         <Table striped={true}>
          <Table.Head>
            <Table.HeadCell>
              Name
            </Table.HeadCell>
            <Table.HeadCell>
              Email
            </Table.HeadCell>
          </Table.Head>
          <Table.Body className="divide-y">
            {attendees.map((attendee) => {
              return (<Table.Row key={attendee._id} className="bg-white dark:border-gray-700 dark:bg-gray-800">
                <Table.Cell className="whitespace-nowrap font-medium text-gray-900 dark:text-white">
                  {attendee.name}
                </Table.Cell>
                <Table.Cell>
                  {attendee.email}
                </Table.Cell>
              </Table.Row>);
            })}
          </Table.Body>
        </Table>
        </div>
        :
        <div>
          <div>
            <p>Attendees and Invitees</p>
            <div className="mt-1 flex flex-row">
              <Dropdown
                label={"Filter"}
                className="bg-gray-50"
                dismissOnClick={false}
              >
                <Dropdown.Item>
                  <input id="checkbox-item-1" type="checkbox" checked={checkedArray[0]} onChange={(e) => setCheckedArray([!checkedArray[0], checkedArray[1], checkedArray[2]])} className="w-4 h-4 text-blue-600 bg-gray-100 border-gray-300 rounded focus:ring-blue-500 dark:focus:ring-blue-600 dark:ring-offset-gray-700 dark:focus:ring-offset-gray-700 focus:ring-2 dark:bg-gray-600 dark:border-gray-500 p-2" />
                  <span className="pl-2">Attending and Invited</span>
                </Dropdown.Item>
                <Dropdown.Item>
                  <input id="checkbox-item-1" type="checkbox" checked={checkedArray[1]} onChange={(e) => setCheckedArray([checkedArray[0], !checkedArray[1], checkedArray[2]])} className="w-4 h-4 text-blue-600 bg-gray-100 border-gray-300 rounded focus:ring-blue-500 dark:focus:ring-blue-600 dark:ring-offset-gray-700 dark:focus:ring-offset-gray-700 focus:ring-2 dark:bg-gray-600 dark:border-gray-500 p-2" />
                  <span className="pl-2">Not Attending and Invited</span>
                </Dropdown.Item>
                <Dropdown.Item>
                  <input id="checkbox-item-1" type="checkbox" checked={checkedArray[2]} onChange={(e) => setCheckedArray([checkedArray[0], checkedArray[1], !checkedArray[2]])} className="w-4 h-4 text-blue-600 bg-gray-100 border-gray-300 rounded focus:ring-blue-500 dark:focus:ring-blue-600 dark:ring-offset-gray-700 dark:focus:ring-offset-gray-700 focus:ring-2 dark:bg-gray-600 dark:border-gray-500 p-2" />
                  <span className="pl-2">Attending and Not Invited</span>
                </Dropdown.Item>
              </Dropdown>
            </div>
          </div>
         <Table striped={true}>
          <Table.Head>
            <Table.HeadCell>
              Name
            </Table.HeadCell>
            <Table.HeadCell>
              Email
            </Table.HeadCell>
            <Table.HeadCell>
              Attending?
            </Table.HeadCell>
            <Table.HeadCell>
              Invited?
            </Table.HeadCell>
          </Table.Head>
          <Table.Body className="divide-y">
            {attendeesAndInvitees
            .filter((person) => toShow(person))
            .map((person) => {
              return (<Table.Row key={person._id} className="bg-white dark:border-gray-700 dark:bg-gray-800">
                <Table.Cell className="whitespace-nowrap font-medium text-gray-900 dark:text-white">
                  {person.name}
                </Table.Cell>
                <Table.Cell>
                  {person.email}
                </Table.Cell>
                <Table.Cell>
                  {person.attending.includes(event._id) ? 'Yes' : 'No'}
                </Table.Cell>
                <Table.Cell>
                  {person.invited.includes(event._id) ? 'Yes' : 'No'}
                </Table.Cell>
              </Table.Row>);
            })}
          </Table.Body>
        </Table>
        </div>
        }
       </div>
     </div>
   </div>
 );
}
