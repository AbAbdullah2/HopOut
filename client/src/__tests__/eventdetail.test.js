import '@testing-library/jest-dom'
import renderer from 'react-test-renderer';
import EventDetail from '../pages/EventList'

const mockedUsedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
   ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate,
}));
Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation(query => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

// test('matches snapshot', async () => {
//   const tree = await renderer.create(<EventDetail/>).toJSON();
//   expect(tree).toMatchSnapshot();
// });