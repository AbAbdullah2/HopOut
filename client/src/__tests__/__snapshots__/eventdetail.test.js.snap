// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="bg-stone-100 min-h-screen"
>
  <div
    className="mx-auto flex flex-col items-center justify-center h-full"
  >
    <div
      className="w-full bg-slate-800 p-5 text-white flex flex-row flex-nowrap justify-between"
    >
      <a
        className="mx-2"
        href="/"
      >
        <span
          className="text-4xl font-extrabold italic text-blue-400 mr-1"
        >
          H
        </span>
        <span
          className="text-4xl"
        >
          opOut
        </span>
      </a>
      <div
        className="flex justify-end content-end items-end right-0 text-2xl font-semibold space-x-4 mx-2 mr-4"
      >
        <a
          className="hover:text-blue-400"
          href="/addFriends"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-user-group px-1"
            data-icon="user-group"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 640 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M96 128a128 128 0 1 1 256 0A128 128 0 1 1 96 128zM0 482.3C0 383.8 79.8 304 178.3 304h91.4C368.2 304 448 383.8 448 482.3c0 16.4-13.3 29.7-29.7 29.7H29.7C13.3 512 0 498.7 0 482.3zM609.3 512H471.4c5.4-9.4 8.6-20.3 8.6-32v-8c0-60.7-27.1-115.2-69.8-151.8c2.4-.1 4.7-.2 7.1-.2h61.4C567.8 320 640 392.2 640 481.3c0 17-13.8 30.7-30.7 30.7zM432 256c-31 0-59-12.6-79.3-32.9C372.4 196.5 384 163.6 384 128c0-26.8-6.6-52.1-18.3-74.3C384.3 40.1 407.2 32 432 32c61.9 0 112 50.1 112 112s-50.1 112-112 112z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <span
            className="pl-1 invisible hidden md:visible md:inline"
          >
            Users
          </span>
        </a>
        <a
          className="hover:text-blue-400"
          href="/events"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-calendar px-1"
            data-icon="calendar"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M96 32V64H48C21.5 64 0 85.5 0 112v48H448V112c0-26.5-21.5-48-48-48H352V32c0-17.7-14.3-32-32-32s-32 14.3-32 32V64H160V32c0-17.7-14.3-32-32-32S96 14.3 96 32zM448 192H0V464c0 26.5 21.5 48 48 48H400c26.5 0 48-21.5 48-48V192z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <span
            className="pl-1 invisible hidden md:visible md:inline"
          >
            Events
          </span>
        </a>
        <a
          className="hover:text-blue-400"
          href="/create"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-plus px-1"
            data-icon="plus"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M240 80c0-17.7-14.3-32-32-32s-32 14.3-32 32V224H32c-17.7 0-32 14.3-32 32s14.3 32 32 32H176V432c0 17.7 14.3 32 32 32s32-14.3 32-32V288H384c17.7 0 32-14.3 32-32s-14.3-32-32-32H240V80z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <span
            className="pl-1 invisible hidden md:visible md:inline"
          >
            Create
          </span>
        </a>
        <div>
          <div
            className="w-fit"
            data-testid="flowbite-tooltip-target"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onPointerDown={[Function]}
          >
            <button
              className="flex items-center"
            >
              <div
                className="hover:text-blue-400"
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-bell px-1"
                  data-icon="bell"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M224 0c-17.7 0-32 14.3-32 32V51.2C119 66 64 130.6 64 208v18.8c0 47-17.3 92.4-48.5 127.6l-7.4 8.3c-8.4 9.4-10.4 22.9-5.3 34.4S19.4 416 32 416H416c12.6 0 24-7.4 29.2-18.9s3.1-25-5.3-34.4l-7.4-8.3C401.3 319.2 384 273.9 384 226.8V208c0-77.4-55-142-128-156.8V32c0-17.7-14.3-32-32-32zm45.3 493.3c12-12 18.7-28.3 18.7-45.3H224 160c0 17 6.7 33.3 18.7 45.3s28.3 18.7 45.3 18.7s33.3-6.7 45.3-18.7z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
                <span
                  className="pl-1 invisible hidden md:visible md:inline"
                />
              </div>
            </button>
          </div>
          <div
            className="z-10 w-fit rounded divide-y divide-gray-100 shadow transition-opacity duration-100 invisible opacity-0 border border-gray-200 bg-white text-gray-900 dark:border-none dark:bg-gray-700 dark:text-white"
            data-testid="flowbite-tooltip"
            id=":r0:"
            role="tooltip"
            style={
              Object {
                "left": " ",
                "position": "absolute",
                "top": " ",
              }
            }
            tabIndex={-1}
          >
            <div
              className="py-1 text-sm text-gray-700 dark:text-gray-200"
            >
              <ul
                className="py-1"
              >
                <li
                  className="flex items-center justify-start py-2 px-4 text-sm text-gray-700 cursor-pointer hover:bg-gray-100 dark:text-gray-200 dark:hover:bg-gray-600 dark:hover:text-white"
                  onClick={[Function]}
                >
                  No notifications!
                </li>
              </ul>
            </div>
          </div>
        </div>
        <div
          className="w-fit"
          data-testid="flowbite-tooltip-target"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onPointerDown={[Function]}
        >
          <button
            className="flex items-center"
          >
            <div
              className="hover:text-blue-400"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-user px-1"
                data-icon="user"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M224 256A128 128 0 1 0 224 0a128 128 0 1 0 0 256zm-45.7 48C79.8 304 0 383.8 0 482.3C0 498.7 13.3 512 29.7 512H418.3c16.4 0 29.7-13.3 29.7-29.7C448 383.8 368.2 304 269.7 304H178.3z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
              <span
                className="pl-1 invisible hidden md:visible md:inline"
              >
                Account
              </span>
            </div>
          </button>
        </div>
        <div
          className="z-10 w-fit rounded divide-y divide-gray-100 shadow transition-opacity duration-100 invisible opacity-0 border border-gray-200 bg-white text-gray-900 dark:border-none dark:bg-gray-700 dark:text-white"
          data-testid="flowbite-tooltip"
          id=":r2:"
          role="tooltip"
          style={
            Object {
              "left": " ",
              "position": "absolute",
              "top": " ",
            }
          }
          tabIndex={-1}
        >
          <div
            className="py-1 text-sm text-gray-700 dark:text-gray-200"
          >
            <ul
              className="py-1"
            >
              <li
                className="flex items-center justify-start py-2 px-4 text-sm text-gray-700 cursor-pointer hover:bg-gray-100 dark:text-gray-200 dark:hover:bg-gray-600 dark:hover:text-white"
                onClick={[Function]}
              >
                <a
                  href="/account"
                >
                  Account
                </a>
              </li>
              <li
                className="flex items-center justify-start py-2 px-4 text-sm text-gray-700 cursor-pointer hover:bg-gray-100 dark:text-gray-200 dark:hover:bg-gray-600 dark:hover:text-white"
                onClick={[Function]}
              >
                <a
                  href="/"
                >
                  Sign out
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
    <div
      className="mt-5 w-11/12 flex flex-row flex-nowrap justify-between"
    >
      <div>
        <button
          aria-checked={false}
          className="bg-gray-400 relative inline-flex h-6 w-11 items-center rounded-full"
          data-headlessui-state=""
          id="headlessui-switch-:r4:"
          onClick={[Function]}
          onKeyPress={[Function]}
          onKeyUp={[Function]}
          role="switch"
          tabIndex={0}
          type="button"
        >
          <span
            className="translate-x-1 inline-block h-4 w-4 transform rounded-full bg-white transition"
          />
        </button>
        <span
          className="pl-2"
        >
          Toggle List View
        </span>
      </div>
      <div
        className="justify-end content-end items-end right-0"
      >
        <div
          className="flex flew-row flex-nowrap"
        >
          <div
            className="mr-4"
          >
            <div
              className="w-fit"
              data-testid="flowbite-tooltip-target"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onPointerDown={[Function]}
            >
              <button
                className="text-white bg-blue-700 border border-transparent hover:bg-blue-800 focus:ring-4 focus:ring-blue-300 disabled:hover:bg-blue-700 dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800 dark:disabled:hover:bg-blue-600 focus:!ring-2 group flex h-min items-center justify-center p-0.5 text-center font-medium focus:z-10 rounded-lg"
                disabled={false}
                type="button"
              >
                <span
                  className="flex items-center rounded-md text-sm px-4 py-2"
                >
                  Categories
                  <svg
                    aria-hidden="true"
                    className="ml-2 h-4 w-4"
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    strokeWidth="2"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M19 9l-7 7-7-7"
                      strokeLinecap="round"
                      strokeLinejoin="round"
                    />
                  </svg>
                </span>
              </button>
            </div>
            <div
              className="z-10 w-fit rounded divide-y divide-gray-100 shadow transition-opacity duration-100 invisible opacity-0 border border-gray-200 bg-white text-gray-900 dark:border-none dark:bg-gray-700 dark:text-white"
              data-testid="flowbite-tooltip"
              id=":r5:"
              role="tooltip"
              style={
                Object {
                  "left": " ",
                  "position": "absolute",
                  "top": " ",
                }
              }
              tabIndex={-1}
            >
              <div
                className="py-1 text-sm text-gray-700 dark:text-gray-200"
              >
                <ul
                  className="py-1"
                >
                  <li
                    className="flex items-center justify-start py-2 px-4 text-sm text-gray-700 cursor-pointer hover:bg-gray-100 dark:text-gray-200 dark:hover:bg-gray-600 dark:hover:text-white"
                    onClick={[Function]}
                  >
                    <input
                      checked={false}
                      className="w-4 h-4 text-blue-600 bg-gray-100 border-gray-300 rounded focus:ring-blue-500 dark:focus:ring-blue-600 dark:ring-offset-gray-700 dark:focus:ring-offset-gray-700 focus:ring-2 dark:bg-gray-600 dark:border-gray-500 p-2"
                      id="checkbox-item-1"
                      onChange={[Function]}
                      type="checkbox"
                    />
                    <span
                      className="pl-2"
                    >
                      academics
                    </span>
                  </li>
                  <li
                    className="flex items-center justify-start py-2 px-4 text-sm text-gray-700 cursor-pointer hover:bg-gray-100 dark:text-gray-200 dark:hover:bg-gray-600 dark:hover:text-white"
                    onClick={[Function]}
                  >
                    <input
                      checked={false}
                      className="w-4 h-4 text-blue-600 bg-gray-100 border-gray-300 rounded focus:ring-blue-500 dark:focus:ring-blue-600 dark:ring-offset-gray-700 dark:focus:ring-offset-gray-700 focus:ring-2 dark:bg-gray-600 dark:border-gray-500 p-2"
                      id="checkbox-item-1"
                      onChange={[Function]}
                      type="checkbox"
                    />
                    <span
                      className="pl-2"
                    >
                      sports
                    </span>
                  </li>
                  <li
                    className="flex items-center justify-start py-2 px-4 text-sm text-gray-700 cursor-pointer hover:bg-gray-100 dark:text-gray-200 dark:hover:bg-gray-600 dark:hover:text-white"
                    onClick={[Function]}
                  >
                    <input
                      checked={false}
                      className="w-4 h-4 text-blue-600 bg-gray-100 border-gray-300 rounded focus:ring-blue-500 dark:focus:ring-blue-600 dark:ring-offset-gray-700 dark:focus:ring-offset-gray-700 focus:ring-2 dark:bg-gray-600 dark:border-gray-500 p-2"
                      id="checkbox-item-1"
                      onChange={[Function]}
                      type="checkbox"
                    />
                    <span
                      className="pl-2"
                    >
                      arts
                    </span>
                  </li>
                  <li
                    className="flex items-center justify-start py-2 px-4 text-sm text-gray-700 cursor-pointer hover:bg-gray-100 dark:text-gray-200 dark:hover:bg-gray-600 dark:hover:text-white"
                    onClick={[Function]}
                  >
                    <input
                      checked={false}
                      className="w-4 h-4 text-blue-600 bg-gray-100 border-gray-300 rounded focus:ring-blue-500 dark:focus:ring-blue-600 dark:ring-offset-gray-700 dark:focus:ring-offset-gray-700 focus:ring-2 dark:bg-gray-600 dark:border-gray-500 p-2"
                      id="checkbox-item-1"
                      onChange={[Function]}
                      type="checkbox"
                    />
                    <span
                      className="pl-2"
                    >
                      fun
                    </span>
                  </li>
                  <li
                    className="flex items-center justify-start py-2 px-4 text-sm text-gray-700 cursor-pointer hover:bg-gray-100 dark:text-gray-200 dark:hover:bg-gray-600 dark:hover:text-white"
                    onClick={[Function]}
                  >
                    <input
                      checked={false}
                      className="w-4 h-4 text-blue-600 bg-gray-100 border-gray-300 rounded focus:ring-blue-500 dark:focus:ring-blue-600 dark:ring-offset-gray-700 dark:focus:ring-offset-gray-700 focus:ring-2 dark:bg-gray-600 dark:border-gray-500 p-2"
                      id="checkbox-item-1"
                      onChange={[Function]}
                      type="checkbox"
                    />
                    <span
                      className="pl-2"
                    >
                      other
                    </span>
                  </li>
                </ul>
              </div>
            </div>
          </div>
          <div
            className="mr-2"
          >
            <div
              className="w-fit"
              data-testid="flowbite-tooltip-target"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onPointerDown={[Function]}
            >
              <button
                className="text-white bg-blue-700 border border-transparent hover:bg-blue-800 focus:ring-4 focus:ring-blue-300 disabled:hover:bg-blue-700 dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800 dark:disabled:hover:bg-blue-600 focus:!ring-2 group flex h-min items-center justify-center p-0.5 text-center font-medium focus:z-10 rounded-lg"
                disabled={false}
                type="button"
              >
                <span
                  className="flex items-center rounded-md text-sm px-4 py-2"
                >
                  Event Status
                  <svg
                    aria-hidden="true"
                    className="ml-2 h-4 w-4"
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    strokeWidth="2"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M19 9l-7 7-7-7"
                      strokeLinecap="round"
                      strokeLinejoin="round"
                    />
                  </svg>
                </span>
              </button>
            </div>
            <div
              className="z-10 w-fit rounded divide-y divide-gray-100 shadow transition-opacity duration-100 invisible opacity-0 border border-gray-200 bg-white text-gray-900 dark:border-none dark:bg-gray-700 dark:text-white"
              data-testid="flowbite-tooltip"
              id=":r7:"
              role="tooltip"
              style={
                Object {
                  "left": " ",
                  "position": "absolute",
                  "top": " ",
                }
              }
              tabIndex={-1}
            >
              <div
                className="py-1 text-sm text-gray-700 dark:text-gray-200"
              >
                <ul
                  className="py-1"
                >
                  <li
                    className="flex items-center justify-start py-2 px-4 text-sm text-gray-700 cursor-pointer hover:bg-gray-100 dark:text-gray-200 dark:hover:bg-gray-600 dark:hover:text-white"
                    onClick={[Function]}
                  >
                    <input
                      checked={false}
                      className="w-4 h-4 text-blue-600 bg-gray-100 border-gray-300 rounded focus:ring-blue-500 dark:focus:ring-blue-600 dark:ring-offset-gray-700 dark:focus:ring-offset-gray-700 focus:ring-2 dark:bg-gray-600 dark:border-gray-500 p-2"
                      id="checkbox-item-1"
                      onChange={[Function]}
                      type="checkbox"
                    />
                    <span
                      className="pl-2"
                    >
                      attending
                    </span>
                  </li>
                  <li
                    className="flex items-center justify-start py-2 px-4 text-sm text-gray-700 cursor-pointer hover:bg-gray-100 dark:text-gray-200 dark:hover:bg-gray-600 dark:hover:text-white"
                    onClick={[Function]}
                  >
                    <input
                      checked={false}
                      className="w-4 h-4 text-blue-600 bg-gray-100 border-gray-300 rounded focus:ring-blue-500 dark:focus:ring-blue-600 dark:ring-offset-gray-700 dark:focus:ring-offset-gray-700 focus:ring-2 dark:bg-gray-600 dark:border-gray-500 p-2"
                      id="checkbox-item-1"
                      onChange={[Function]}
                      type="checkbox"
                    />
                    <span
                      className="pl-2"
                    >
                      invited
                    </span>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="my-3 mb-5 w-11/12 items-center justify-center"
    />
  </div>
</div>
`;
